### PR TITLE
Add data dictionary link to mobile nav and make parent project name full-width

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/12485
https://github.com/cityofaustin/atd-data-tech/issues/12377

This updates the mobile navigation with the data dictionary link. I might have gone a bit overboard on reusing the config for both and handling internal vs. external links, but I wanted to give it a shot. I also fixed the issue with the parent project name link not using the full width available to it.

## Testing
**URL to test:** 

https://deploy-preview-1077--atd-moped-main.netlify.app/

**Steps to test:**
1. Test the navigation bar help links in mobile and desktop widths to make sure they still go to the right places
2. Add a parent project to a project and then update the parent project's name to something really long like `Demo Project #1 Demo Project #1 Demo Project #1 Demo Project #1 Demo Project #1 Demo Project #1 Demo Project #1 Demo Project #1 Demo Project #1 Demo Project #1 Demo Project #1 Demo Project #1`. It should take up the full width available to it and wrap below.

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/layouts/DashboardLayout/NavBar/DropdownMenu.js [hunk 1]
```diff
@@ -1,33 +1,59 @@
...
 import { getSessionDatabaseData, useUser } from "../../../auth/user";
 import { getInitials } from "src/utils/userNames";
 import emailToInitials from "../../../utils/emailToInitials";
-import clsx from "clsx";
 
+/**
+ * Configuration for help menu items we iterate to render menu items in DropdownMenu and MobileDropdownMenu
+ * @property {string} linkType - "internal" or "external" to determine how to render link
+ * @property {string} href - URL to link to if linkType is "external"
+ * @property {string} link - URL to link to if linkType is "internal"
```
Comment (mddilley) [inline moped-editor/src/layouts/DashboardLayout/NavBar/DropdownMenu.js:26]: Maybe these two can be merged?

**Review by frankhereford — APPROVED** [review]
I read over this code and checked it out of netlify with your testing steps. This is great, thank you Mike! 

To your question about the `href` and `link` properties -- I do think they at least have confusing names and probably could be merged, but I am going to leave that up to you and everyone with more moped/react experience.

File: moped-editor/src/layouts/DashboardLayout/NavBar/DropdownMenu.js (hunk 1 above)
Comment (johnclary) [reply moped-editor/src/layouts/DashboardLayout/NavBar/DropdownMenu.js:26]: yep, i think so!

**Review by johnclary — APPROVED** [review]
@mddilley this looks great! thanks for catching that issue with the mobile nav.
good stuff!  🚢🍫

**Review by roseeichelmann — APPROVED** [review]
Looks great! Test steps checked out 🚀🚀

**Comment by mddilley** [comment]
Thanks, y'all! I pushed an update to merge the `href` and `link` keys together. I'm resolving one remaining console warning 

```
Warning: Each child in a list should have a unique "key" prop.
Check the render method of `MobileDropdownMenu`.
```
which is odd because I made the same update to both `DropdownMenu` and `MobileDropdownMenu` to squash that warning but it only fixed one of them. Seems related to the conditional rendering so I'll get to the bottom of it.

**Comment by mddilley** [comment]
Got to the bottom of that warning. I was using `item.link` for the key for rendering from the `navigationItems` array. That array has objects that have `href` keys not `link`. So, the key prop for all of the `MenuItem`s rendered for that array was being set an `undefined` resulting in the warning in my last comment. 😵‍💫 

It is sorted out now! Merging.

**Pull request merged**